### DISCUSSION TITLE
cuelang range check

### DISCUSSION THREAD

**Discussion by rgaoistheone** [cue-lang/cue#1805] [Q&A] [answered]
Hey guys, I'm trying to add some range check for my manifest, tried below but no luck 

```replicas: *1 | >= 0 & < 10```

When I pass replicas with value 11, i got errors like ` empty disjunction: conflicting values 1 and 11 (and 3 more errors).`

Any help will be much appreciated !

**Comment by verdverm** [comment]
It would be helpful to know what version and command you are running. https://github.com/cue-lang/cue/wiki/Creating-test-or-performance-reproducers is the preferred way to share reproducers when you have issues or questions. In this case, it would be helpful to know what you expect to happen. 

For example, I am getting a similar but different error message with v0.4.3 as well as on the playground: https://cuelang.org/play/?id=CYqkqKVBEEY#cue@export@cue

**Comment by antong** [comment, marked as answer]
I guess that what you try actually works as you want, but that the error message may be a bit difficult. If I understand what you do, you want to have a range check to say that replicas must be in the range [0, ..., 10). If you set replicas to say any integer from 0 to 9, then you get no error. 11 is not in the range, so you get an error. If you don't set replicas at all, then you get the default, 1.

Note that you are not requiring replicas to be an integer, so you could set it to 3.14 without an error. Perhaps consider:
```
replicas: int | *1
replicas: >= 0 & < 10

//replicas: 0 // -> OK
//replicas: 3.14 // -> error
//replicas: 7 // -> OK
//replicas: 10 // -> error```

**Reply by rgaoistheone** [reply]
Thanks so much ! That's EXACTLY what I want .
```
replicas: int | *1
replicas: >= 0 & < 10
```
This fomula is quite fresh , at least for me :D .
